platform/persistence/platform-persistence-core-listeners/src/main/java/org/codice/ddf/persistence/events/AlertListener.java

```diff
@@ -0,0 +1,189 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.persistence.events;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.codice.ddf.system.alerts.Alert;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * AlertListener is a kind of Decanter alerter. It listens for alerts on the decanter/alert/* topic
+ * and persists the alerts to solr. Alerts from the same host and source are merged into one solr
+ * entry.
+ */
+public class AlertListener implements EventHandler {
+
+    private static final String CORE_NAME = "alerts";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AlertListener.class);
+
+    private PersistentStore persistentStore;
+
+    private String hostName;
+
+    public AlertListener(PersistentStore persistentStore) {
+        this.persistentStore = persistentStore;
+    }
+
+    public void init() throws UnknownHostException {
+        this.hostName = InetAddress.getLocalHost()
+                .getHostName();
+    }
+
+    /**
+     * Processes alert events and stores them in the persistent store. Because this handler 'squashes'
+     * alerts into one entry it needs to be synchronized to be thread safe.
+     *
+     * @param event The osgi alert event
+     * @throws IllegalArgumentException
+     */
+    @Override
+    public synchronized void handleEvent(Event event) throws IllegalArgumentException {
+        if (Alert.ALERT_DISMISS_TOPIC.equals(event.getTopic())) {
+            dismissAlert(event);
+            return;
+        }
+        //need source to lookup previous alerts
+        if (event.getProperty(Alert.SYSTEM_NOTICE_SOURCE_KEY) == null) {
+            return;
+        }
+        LOGGER.debug("Received alert event on topic {}", event.getTopic());
+
+        Alert alert = getAlertFromEvent(event);
+        Alert existingAlert = getAlertBySource(alert.getSource());
+        if (existingAlert != null) {
+            alert.setId(existingAlert.getId());
+            alert.setCount(existingAlert.getCount() + 1);
+            alert.setTime(existingAlert.getTime());
+        }
+
+        addAlertToStore(alert);
+    }
+
+    private void addAlertToStore(Alert alert) {
+        PersistentItem item = new PersistentItem();
+        item.addIdProperty(alert.getId());
+        item.addProperty(Alert.SYSTEM_NOTICE_SOURCE_KEY, alert.getSource());
```

Comment:
❓ Could we loop over the map entries returned by `Alert.getProperties()` and not have to worry about all the constants and adding every single property one by one? Another option could be to move the `SystemNotice` and `Alert` classes in the `platform` layer so that we can add a `Alert.toPersistentItem()` method and have all this logic inside the `Alert` class.